### PR TITLE
Add missing datasets to public API

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Description
@scikit-image/core I noticed a few datasets were missing from `data.__init__.py`.

I added them and updated the examples that used them.

However, I am unclear about the exact meaning of these datasets, and their importance. 

Help is appreciated elaborating on the description of each dataset.

TODO:
- [x] Provide correct context https://github.com/scikit-image/scikit-image/pull/4951#discussion_r480111905
- [ ] data called `cells3d`
- [ ] data called `mitosis`

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: doc/examples/applications/plot_3d_image_processing.py
Comment (alexdesiqueira) [inline doc/examples/applications/plot_3d_image_processing.py:53]: dependency* :slightly_smiling_face:

File: skimage/data/__init__.py
Comment (alexdesiqueira) [inline skimage/data/__init__.py:622]: 1 space only, maybe?

**Comment by alexdesiqueira** [comment]
> However, I am unclear about the exact meaning of these datasets, and their importance.
> Help is appreciated elaborating on the description of each dataset.

Not sure what you mean here @hmaarrfk. Where would we describe? What would you like to see?
There are some descriptions on our [GitLab data repo](https://gitlab.com/scikit-image/data/). Is this useful, somehow?

File: skimage/data/__init__.py
Comment (alexdesiqueira) [inline skimage/data/__init__.py:598]: I think information like 598–609 should be in a `Notes` section.

Comment (hmaarrfk) [reply skimage/data/__init__.py:598]: Understanding your data is critical to good analysis and visualization.

What is the difference between the "Notes" and the "Description".

If you feel strongly about it, do reorganize.

**Comment by hmaarrfk** [comment]
Thanks for the review @alexdesiqueira! I don't see the need for a Notes section if the description is empty.

IMO: we can reorganize once the description gets crowded.

File: skimage/data/__init__.py
Comment (alexdesiqueira) [reply skimage/data/__init__.py:598]: We follow the [Numpy docstyle](https://numpydoc.readthedocs.io/en/latest/format.html). According to them:
> The sections of a function’s docstring are:
>  `1.` Short summary
>  A one-line summary that does not use variable names or the function name (...)
> (...)
> `13.` Notes
> An optional section that provides additional information about the code, possibly including a discussion (...)

Comment (hmaarrfk) [reply skimage/data/__init__.py:598]: I've made some modifications, however, I will point to:

> Extended Summary
> 
> A few sentences giving an extended description. This section should be used to clarify functionality, not to discuss implementation detail or background theory, which should rather be explored in the Notes section below. You may refer to the parameters and the function name, but parameter descriptions still belong in the Parameters section.

Functionality here, for me, represents things like metadata, depicting voxel size, which are simply not communicated in a numpy array. I've kept those details.

**Comment by sciunto** [comment]
CC @mkcor

File: doc/examples/applications/plot_3d_image_processing.py
Comment (sciunto) [inline doc/examples/applications/plot_3d_image_processing.py:52]:
```suggestion
# ==========================
```

File: doc/examples/applications/plot_3d_image_processing.py
Comment (sciunto) [inline doc/examples/applications/plot_3d_image_processing.py:54]: This line is not necessary.

File: skimage/data/__init__.py
Comment (sciunto) [inline skimage/data/__init__.py:634]: This blank line can be deleted.

File: skimage/data/__init__.py
Comment (sciunto) [inline skimage/data/__init__.py:645]:
```suggestion
           124(6):1283-98 / :DOI:`10.1016/j.cell.2006.01.040` PMID 16564017
```

File: skimage/data/__init__.py
Comment (sciunto) [inline skimage/data/__init__.py:635]:
```suggestion
    Copyright David Root. Licensed under CC-0 [2]_.
```

**Comment by sciunto** [comment]
I think my comments will turn the CI green. Once applied, i'm ready to approve.

**Review by jni — APPROVED** [review]
Approved with @sciunto's suggestions.

File: skimage/data/tests/test_data.py
Comment (mkcor) [inline skimage/data/tests/test_data.py:154]: Looks like this should be updated to reflect the changes in this PR.

File: skimage/data/__init__.py
Comment (mkcor) [inline skimage/data/__init__.py:625]:
```suggestion
    """Image of human cells undergoing mitosis.
```

File: skimage/data/__init__.py
Comment (mkcor) [reply skimage/data/__init__.py:626]: @alexdesiqueira personally I don't understand this suggestion ^^

File: skimage/data/__init__.py
Comment (mkcor) [reply skimage/data/__init__.py:635]: Regarding all the metadata, maybe we could point to the `AS_09125_050116030001_D03f00d0.tif` entry at https://gitlab.com/scikit-image/data/-/blob/master/README.md#data in a 'single' source perspective?

File: skimage/data/__init__.py
Comment (hmaarrfk) [reply skimage/data/__init__.py:626]: I think it is outdated.

File: doc/examples/applications/plot_3d_image_processing.py
Comment (hmaarrfk) [reply doc/examples/applications/plot_3d_image_processing.py]: Thanks

File: doc/examples/applications/plot_3d_image_processing.py
Comment (hmaarrfk) [reply doc/examples/applications/plot_3d_image_processing.py]: Thanks

File: skimage/data/__init__.py
Comment (hmaarrfk) [reply skimage/data/__init__.py:645]: thanks

File: skimage/data/__init__.py
Comment (hmaarrfk) [reply skimage/data/__init__.py:634]: thanks

File: skimage/data/__init__.py
Comment (hmaarrfk) [reply skimage/data/__init__.py:635]: I would rather copy over what is missing. Do you want me to point to anything else that you've included in the README?

File: skimage/data/tests/test_data.py
Comment (hmaarrfk) [reply skimage/data/tests/test_data.py:154]: thanks!

File: skimage/data/__init__.py
Comment (hmaarrfk) [reply skimage/data/__init__.py:625]: Thanks

**Comment by hmaarrfk** [comment]
Thank you all for the reviews.

@mkcor suggested that we unify the names in the registry with the names of the functions.

Now that I look at this, I notice that we might be "breaking" a private API. Are you guys OK with that? I'm not sure how many people you've pointed to these functions.

File: skimage/data/__init__.py
Comment (mkcor) [reply skimage/data/__init__.py]: Ok. I didn't mean that anything was missing (more like the opposite).

Maybe this sentence isn't correct: "Data of human cells undergoing mitosis taken from [1]_." Strictly speaking, the image wasn't taken from that paper, it was taken from CellProfiler as detailed here: https://scikit-image.org/docs/dev/auto_examples/applications/plot_human_mitosis.html#segment-human-cells-in-mitosis
(The image is part of the data used in the cited paper, but does not ship _as is_ with the paper, even considering the supplemental material, etc.)

**Comment by mkcor** [comment]
> @mkcor suggested that we unify the names in the registry with the names of the functions.

Did I? ;-)

Are you referring to my [considerations](https://github.com/scikit-image/scikit-image/pull/4939#issuecomment-679869596) about one-word names?

File: skimage/data/__init__.py
Comment (hmaarrfk) [reply skimage/data/__init__.py]: hmm, thanks for the details.

I think the context is very important. I'll try to add that phrase in.

That said, renaming the fiels in registry broke some tests too. so this will take a little longer than i had hoped for today.

**Comment by hmaarrfk** [comment]
I'm refering to https://github.com/scikit-image/scikit-image/pull/4951#discussion_r479980826  :D

**Comment by mkcor** [comment]
> I'm refering to [#4951 (comment)](https://github.com/scikit-image/scikit-image/pull/4951#discussion_r479980826) :D

Oh! Well, if this PR is merged, `fetch('data/cells.tif')` simply won't work anymore... So this comment of mine wasn't discussing naming conventions.

But :wink: I'm always happy to have a discussion about naming conventions, as @jni knows! I know that we already have `cell` in our data registry, but I'm not too fond of `cells3D`. Then, should we make sure all 3D images have a name ending with '3D'...? I prefer documenting this in the docstring.

File: skimage/data/__init__.py
Comment (mkcor) [reply skimage/data/__init__.py]: Maybe they don't need renaming...? :p

**Comment by hmaarrfk** [comment]
Honestly, i don't know what to call this dataset.

Context would help. I would prefer not naming it `cells3D` either.

Something more descriptive of the experiment would be really nice.

File: skimage/data/__init__.py
Comment (alexdesiqueira) [reply skimage/data/__init__.py]: @mkcor I meant two blank lines -> one blank one :slightly_smiling_face: Wasn't clear enough, sorry about that.

**Comment by alexdesiqueira** [comment]
> Context would help. I would prefer not naming it cells3D either.

In this case, I'd go for "if isn't broke don't fix it" :joy: `cells` is the best way to go (until we find a better one).

**Comment by jni** [comment]
Sorry, I'm not sure about the complete discussion here. Can someone summarise the proposed options? Regarding deprecating names of functions already released, I'd prefer we avoid that, even if we didn't publicise them. Someone might have done `data.<tab>` and used some of those functions. For stuff that hasn't been released, we can be flexible going forward.

On example is the cells-3d dataset. For that one, if we haven't actually published it in the `data.` namespace, I think *actually* we should rework it to contain *both* channels — there is a `membranes.tif` file in the tutorials repo that corresponds to the same cells. As to what to call it... I dunno, I kinda like cells3d (lowercase d), other alternatives welcome...

**Comment by hmaarrfk** [comment]
@mkcor would you mind helping me attribute the data correctly?

Do you have al link to the original data?

> Maybe this sentence isn't correct: "Data of human cells undergoing mitosis taken from [1]_." Strictly speaking, the image wasn't taken from that paper, it was taken from CellProfiler as detailed here: https://scikit-image.org/docs/dev/auto_examples/applications/plot_human_mitosis.html#segment-human-cells-in-mitosis
(The image is part of the data used in the cited paper, but does not ship as is with the paper, even considering the supplemental material, etc.)

File: skimage/data/tests/test_data.py
Comment (mkcor) [inline skimage/data/tests/test_data.py:176]: I don't support this change :wink: (only referring to the EOF thing)

**Comment by mkcor** [comment]
> Sorry, I'm not sure about the complete discussion here. Can someone summarise the proposed options? Regarding deprecating names of functions already released, I'd prefer we avoid that, even if we didn't publicise them. Someone might have done `data.<tab>` and used some of those functions. For stuff that hasn't been released, we can be flexible going forward.

@jni `data.mitosis()` and `data.cells()` have not been published yet; they are indeed [missing](https://github.com/scikit-image/scikit-image/pull/4939#issuecomment-682831131). With this PR, @hmaarrfk is filling this gap while renaming `mitosis` into `human_mitosis` -- I don't particularly support this renaming because, then, should we rename [kidney](https://github.com/scikit-image/scikit-image/blob/73a8dbf0128e8832b15d86c88c9304bc1280aa71/skimage/data/__init__.py#L40) into `mouse_kidney`?

> On example is the cells-3d dataset. For that one, if we haven't actually published it in the `data.` namespace, I think _actually_ we should rework it to contain _both_ channels — there is a `membranes.tif` file in the tutorials repo that corresponds to the same cells. As to what to call it... I dunno, I kinda like cells3d (lowercase d), other alternatives welcome...

Interesting, I wasn't aware of this. Ok, so let's look into this more carefully before we publish `membranes` and/or `cells3d`.

**Comment by mkcor** [comment]
> @mkcor would you mind helping me attribute the data correctly?
> 
> Do you have a link to the original data?

Sure, it is: https://github.com/CellProfiler/examples/blob/master/ExampleHuman/images/AS_09125_050116030001_D03f00d0.tif as documented under `AS_09125_050116030001_D03f00d0.tif` at https://gitlab.com/scikit-image/data/#data

**Comment by mkcor** [comment]
> mouse_kidney

s/mouse/murine/

**Comment by hmaarrfk** [comment]
thanks for the reference for human mitosis.

@jni do you k,now the original link for the cells3d?

**Comment by emmanuelle** [comment]
> @jni do you k,now the original link for the cells3d?

Knowing more about the dataset might help crafting a new name.

**Comment by emmanuelle** [comment]
Actually I'm quite in favor of `cells3d` because it tells user they should not directly call `plt.imshow` on the image array.

File: doc/examples/applications/plot_3d_image_processing.py
Comment (hmaarrfk) [inline doc/examples/applications/plot_3d_image_processing.py:47]:
```suggestion
from skimage.data import cells3d
```

File: doc/examples/applications/plot_3d_image_processing.py
Comment (hmaarrfk) [inline doc/examples/applications/plot_3d_image_processing.py:54]:
```suggestion
data = cells3d()
```

File: doc/examples/color_exposure/plot_adapt_hist_eq_3d.py
Comment (hmaarrfk) [inline doc/examples/color_exposure/plot_adapt_hist_eq_3d.py:27]:
```suggestion
from skimage.data import cells3d
```

File: doc/examples/color_exposure/plot_adapt_hist_eq_3d.py
Comment (hmaarrfk) [inline doc/examples/color_exposure/plot_adapt_hist_eq_3d.py:29]:
```suggestion
im_orig = cells3d()
```

File: skimage/data/__init__.py
Comment (hmaarrfk) [inline skimage/data/__init__.py:597]:
```suggestion
def cells3d():
```

**Comment by hmaarrfk** [comment]
@emmanuelle thanks for chiming in.

The bigger thing i wanted to address was removing the notion of calling fetch from the examples.

If anybody wants to elaborate on the details of the dataset,i won't be offended by direct modifications to the branch ;)

**Comment by emmanuelle** [comment]
As @mkcor I'd rather have `mitosis` than `human_mitosis`, in order to keep short names. Any objection?

File: doc/examples/applications/plot_human_mitosis.py
Comment (jni) [inline doc/examples/applications/plot_human_mitosis.py:29]: We've been discussing "just" mitosis for this but actually, in biological circles, "mitosis" refers to a well known dataset that ships with Fiji, pictured in my napari announcement video:

https://twitter.com/jnuneziglesias/status/1187400066987741186

(a) we should probably add that dataset to skimage, but
(b) let's not use just mitosis. I'd rather "mitosis2d", for example?

Comment (jni) [reply doc/examples/applications/plot_human_mitosis.py:29]: @emmanuelle 👆

Comment (sciunto) [reply doc/examples/applications/plot_human_mitosis.py:29]: > (b) let's not use just mitosis. I'd rather "mitosis2d", for example?


:-1:  we said we add "3d" to confocal images earlier.

Comment (mkcor) [reply doc/examples/applications/plot_human_mitosis.py:29]: > :-1: we said we add "3d" to confocal images earlier.

:thinking: we're appending "3d" because the image is 3D, not necessarily because it's from confocal fluorescence microscopy (unless I missed something). So appending "2d" would be consistent...

Comment (sciunto) [reply doc/examples/applications/plot_human_mitosis.py:29]: yes, but 2D images do not have 2d tags.

Comment (sciunto) [reply doc/examples/applications/plot_human_mitosis.py:29]: I said confocal because so far, we have only confocal images and confocal microscopes provide 3D images.

Comment (sciunto) [reply doc/examples/applications/plot_human_mitosis.py:29]: To clarify, having cell / cell3d and mitosis2d / mistosis, It sounds bad in terms of consistency and given the effort we are putting to fix former choices like this in the library, I prefer to put a veto now :)

Comment (mkcor) [reply doc/examples/applications/plot_human_mitosis.py:29]: > having cell / cell3d

Wait, we have `cell` and (with this PR) `cells3d` (formerly `cells` but unregistered so far) :)

> and mitosis2d / mitosis

I see what you mean now.

**Comment by emmanuelle** [comment]
ok, thanks for the explanations everyone. So is everybody ok for `cells3d`? If yes, there is only the mitosis image to decide.

File: doc/examples/applications/plot_human_mitosis.py
Comment (emmanuelle) [reply doc/examples/applications/plot_human_mitosis.py:29]: @jni but if we integrate the Fiji dataset at some point, can't we call it `mitosis3d` ? Since we document our datasets with specific gallery examples, I don't think it's a big problem that we call mitosis something else than what Fiji does.

Comment (jni) [reply doc/examples/applications/plot_human_mitosis.py:29]: Well, it would be mitosis5d. ;) imho if we can aim for consistency with our neighbours, we should. As pointed out by @mkcor, we don't yet have `cells3d`, so we can try to think of something better that maintains our own consistency.

@sciunto 

> I prefer to put a veto now :)

Veto is such a strong word — maybe prefer to have a robust discussion to arrive at the optimal choice now? =) I agree with you that `cell`/`cells3d` vs `mitosis2d`/`mitosis` sounds bad. Perhaps we can come up with something better within the constraints below. If we can't, *then* we can start to debate whether we should remove some constraints:

- the use of the `nd` suffix should be consistent.
- we should avoid using `fetch` in examples.
- `mitosis` should be used for the "famous" mitosis dataset.

I think possibly the last constraint, "datasets should only have one word", could be relaxed, and then `human_mitosis` can be `mitosis_screen` (it is an example from a high-content imaging screen). 

I'd propose an additional constraint that would make our lives harder, but maybe it can work:

- names should not be prefixes of other names

since this makes autocomplete annoying. For `cells3d`, we could use `fibroblasts` for example (I think that's what they were, otherwise we can adjust 😅).

File: skimage/data/__init__.py
Comment (alexdesiqueira) [inline skimage/data/__init__.py:625]: If we decide to call it `cells3d`, I guess it'd be wise to rename the file as well — it would help to maintain consistency.

**Comment by emmanuelle** [comment]
@jni if I remember correctly, you were going to propose a list of names here. Is it still the plan?

**Comment by jni** [comment]
@emmanuelle that was indeed the plan but I have dropped the ball. To be honest I somehow care less about this than I did at the time — maybe the whole camera saga has taught me that these things can eventually be changed, even if it takes some effort. (Sorry @alexdesiqueira :joy:)

I'm happy to stick with `mitosis` for the `human_mitosis` data.

**Comment by jni** [comment]
But, by the way, I do care that `cells3d` should include both channels. The channels are present as `cells.tif` and `cells_membrane.tif` in the [tutorials repo](https://github.com/scikit-image/skimage-tutorials/tree/master/images)

File: skimage/data/__init__.py
Comment (grlee77) [inline skimage/data/__init__.py:608]:
```suggestion
    cells3d: (60, 256, 256) float64 ndarray
```

**Review by grlee77 — APPROVED** [review]
Thanks @hmaarfk. There is one remaining suggestion regarding renaming `cells.tif` to `cells3d.tif`.

**Comment by hmaarrfk** [comment]
Thanks @grlee77 i've fixed a few more thngs. 

I'm a little hesitant that we are pointing to master, and not to a specific branch or to more permanent links.

That said, the sample data feels more ephemeral to me, than the library itself.

**Comment by grlee77** [comment]
I see there was one other issue by @jni that remains to be addressed: https://github.com/scikit-image/scikit-image/pull/4951#issuecomment-712152185

@jni, was your intention that `skimage.data.cells3d` should return a shape (60, 256, 256, 2) array created by stacking these two files?

Here is a plot of a single slice of the 3D datasets overlaid for context
```Python
from skimage import io
import matplotlib.pyplot as plt

m = io.imread('cells_membrane.tif')
c = io.imread('cells.tif')
fig, axes = plt.subplots(1, 3)
axes[0].imshow(c[30])
axes[0].set_title('cells.tif')
axes[1].imshow(m[30])
axes[1].set_title('cells_membrane.tif')
axes[2].imshow(c[30] + 2*m[30])
axes[2].set_title('overlaid')
```
cells_and_membranes

**Comment by jni** [comment]
Yeah, that's my goal/intention. I intend to push to this branch shortly, hold please!

**Comment by jni** [comment]
Done! Assuming tests pass this is ok from me.

**Comment by jni** [comment]
🎉 🚀

**Comment by hmaarrfk** [comment]
Thanks everybody

**Comment by jni** [comment]
Thank **you**, @hmaarrfk! You always do the hard work and then it sits in committee for months! But we deeply appreciate it! ❤️

**Pull request merged**
